Possible bug demonstrated by Python multihost integration tests in CI

Failure seen at https://github.com/feldera/cloud/actions/runs/25405915204/job/74519490716#step:20:505:

```
[6](https://github.com/feldera/cloud/actions/runs/25405915204/job/74519490716#step:20:318)

=================================== FAILURES ===================================
___________ TestPipeline.test_url_encoding_ingress_egress_table_name ___________
[gw8] linux -- Python 3.10.16 /__w/cloud/cloud/feldera/python/.venv/bin/python

self = <tests.platform.test_shared_pipeline.TestPipeline testMethod=test_url_encoding_ingress_egress_table_name>

    def test_url_encoding_ingress_egress_table_name(self):
        """
        CREATE TABLE "t1#a1" (
            c1 TEXT NOT NULL
        ) WITH ('materialized' = 'true');
        """
        # Test egress with URL encoding - listen directly to table with special characters
        self.pipeline.start_paused()
        out = self.pipeline.listen("t1#a1")
    
        # Test ingress with URL encoding
        self.pipeline.resume()
        data = [{"c1": "test_value"}]
    
        # Test pushing data to table with special characters in name
        TEST_CLIENT.push_to_pipeline(
            pipeline_name=self.pipeline.name,
            table_name="t1#a1",
            format="json",
            array=True,
            data=data,
        )
    
        # Verify data was inserted correctly via query
        result = list(self.pipeline.query('SELECT * FROM "t1#a1"'))
        expected = [{"c1": "test_value"}]
        self.assertCountEqual(result, expected)
    
        # Insert another value to test egress
        additional_data = [{"c1": "test_value_2"}]
        TEST_CLIENT.push_to_pipeline(
            pipeline_name=self.pipeline.name,
            table_name="t1#a1",
            format="json",
            array=True,
            data=additional_data,
        )
    
        # Verify egress data includes both values with insert_delete markers
        egress_result = out.to_dict()
        expected_egress = [
            {"c1": "test_value", "insert_delete": 1},
            {"c1": "test_value_2", "insert_delete": 1},
        ]
>       self.assertCountEqual(egress_result, expected_egress)
E       AssertionError: Element counts were not equal:
E       First has 0, Second has 1:  {'c1': 'test_value_2', 'insert_delete': 1}

tests/platform/test_shared_pipeline.py:787: AssertionError
```

--- a/python/tests/platform/test_shared_pipeline.py
+++ b/python/tests/platform/test_shared_pipeline.py
@@ -779,6 +779,7 @@ class TestPipeline(SharedTestPipeline):
         )
 
         # Verify egress data includes both values with insert_delete markers
+        self.pipeline.wait_for_idle()
         egress_result = out.to_dict()
         expected_egress = [
             {"c1": "test_value", "insert_delete": 1},

